drivers/exec/driver.go

```diff
@@ -402,11 +402,42 @@ func (d *Driver) WaitTask(ctx context.Context, taskID string) (<-chan *drivers.E
 	return ch, nil
 }
 
+// handleFailedExecutor is called when we receive an unrecoverable error from
+// an executor process.
+//
+// Here we attempt to clean up the orphaned subprocesses before returning the
+// error to the Nomad client to avoid orphaning the process and running duplicate
+// tasks.
+func (d *Driver) handleFailedExecutor(handle *taskHandle) error {
```

Comment:
I see why you duplicated this everywhere, but I think we could stick something in a utils package or even executor thats `CleanupPID(pid int) error` then we can just omit this function all together and call `executor.CleanupPID(handle.pid)`.

We don't reference the Driver at all in this func anyways. Thoughts?

Reply:
Yup that makes sense 👍